src/citrine/resources/dataset.py

```diff
@@ -91,6 +92,10 @@ class Dataset(Resource['Dataset']):
     update_time = properties.Optional(properties.Datetime(), 'update_time')
     delete_time = properties.Optional(properties.Datetime(), 'delete_time')
     public = properties.Optional(properties.Boolean(), 'public')
+    project_id = properties.Optional(properties.UUID(), 'project_id',
+                                     serializable=False, deserializable=False)
+    session = properties.Optional(properties.Object(Session), 'session',
+                                  serializable=False, deserializable=False)
```

Comment:
For some reason, these fields were never declared but were used everywhere.

Reply:
Ugh, this is an object model mess.  This was designed so that users would make stubby little datasets that then got registered, and only after registration (which returns a new object) would it have session & other parameters necessary for platform interaction.  Omitting the methods from the constructor was an intentional editorial choice.

Perhaps the cleaner solution is swapping this over to `deserializable=True` and then adding the munge to the Collection `build` method like we've done elsewhere?